test/services/licences/supplementary/determine-licence-flags.service.test.js

```diff
@@ -0,0 +1,117 @@
+      before(async () => {
+        licence = await LicenceHelper.add({ includeInSrocBilling: true, includeInPresrocBilling: 'yes' })
+      })
+
+      describe("and a charging scheme of 'sroc'", () => {
+        before(() => {
+          scheme = 'sroc'
+        })
+
+        it('returns the licence details', async () => {
+          const result = await DetermineLicenceFlagsService.go(licence.id, scheme)
+
+          expect(result.licenceId).to.equal(licence.id)
+          expect(result.regionId).to.equal(licence.regionId)
+          expect(result.startDate).to.equal(null)
+          expect(result.endDate).to.equal(null)
+        })
+
+        it('returns the correct flags', async () => {
+          const result = await DetermineLicenceFlagsService.go(licence.id, scheme)
+
+          expect(result.flagForPreSrocSupplementary).to.equal(true)
+          expect(result.flagForSrocSupplementary).to.equal(true)
+          expect(result.flagForTwoPartTariffSupplementary).to.equal(false)
+        })
+      })
+
+      describe("and a charging scheme of 'alcs'", () => {
+        it('returns the licence details', async () => {
+          const result = await DetermineLicenceFlagsService.go(licence.id, scheme)
+
+          expect(result.licenceId).to.equal(licence.id)
+          expect(result.regionId).to.equal(licence.regionId)
+          expect(result.startDate).to.equal(null)
+          expect(result.endDate).to.equal(null)
+        })
+
+        it('returns the correct flags', async () => {
+          const result = await DetermineLicenceFlagsService.go(licence.id, scheme)
+
+          expect(result.flagForPreSrocSupplementary).to.equal(true)
+          expect(result.flagForSrocSupplementary).to.equal(true)
+          expect(result.flagForTwoPartTariffSupplementary).to.equal(false)
+        })
+      })
+    })
+
+    describe('for a licence with no pre-sroc or sroc flag', () => {
+      before(async () => {
+        licence = await LicenceHelper.add()
+      })
+
+      describe("and a charging scheme of 'sroc'", () => {
+        before(() => {
+          scheme = 'sroc'
+        })
+
+        it('returns the licence details', async () => {
+          const result = await DetermineLicenceFlagsService.go(licence.id, scheme)
+
+          expect(result.licenceId).to.equal(licence.id)
+          expect(result.regionId).to.equal(licence.regionId)
+          expect(result.startDate).to.equal(null)
+          expect(result.endDate).to.equal(null)
+        })
+
+        it('returns the correct flags', async () => {
+          const result = await DetermineLicenceFlagsService.go(licence.id, scheme)
+
+          expect(result.flagForPreSrocSupplementary).to.equal(false)
+          expect(result.flagForSrocSupplementary).to.equal(true)
+          expect(result.flagForTwoPartTariffSupplementary).to.equal(false)
+        })
+      })
+
+      describe("and a charging scheme of 'alcs'", () => {
+        it('returns the licence details', async () => {
```

Comment:
I don't think this test is right as the scheme doesn't appear to have been set to "alcs" in a`before` and the flagging in the `expect` looks wrong.

Reply:
Good spot